src/components/widgets/filesystem/FileEditor.vue

```diff
@@ -1,38 +1,112 @@
 <template>
-  <div
-    ref="monaco-editor"
-    class="editor"
-  >
+  <div class="editor-container d-flex">
     <div
-      v-if="!editor"
-      class="spinner"
+      ref="monaco-editor"
+      :class="(sidebar && !isMobile)?'editor':'editor-full'"
     >
-      <v-progress-circular
-        indeterminate
-        size="100"
-        color="primary"
-      />
+      <div
+        v-if="!editor"
+        class="spinner"
+      >
+        <v-progress-circular
+          indeterminate
+          size="100"
+          color="primary"
+        />
+      </div>
+    </div>
+    <div
+      v-if="sidebar && !isMobile"
+      ref="editor-sidebar"
+      class="editor-sidebar pa-2"
+    >
+      <v-card flat>
+        <v-card-subtitle>
+          <v-icon small>
+            $tree
+          </v-icon>
+          {{ $t('app.file_system.label.structure') }}
+        </v-card-subtitle>
+        <v-divider />
+        <div class="pa-2 heading">
+          <div
+            v-for="(item, index) in structures"
+            :id="'line_'+item.index"
+            :key="index"
+            style="cursor: pointer"
+            :class="(item.section!==undefined?'':'pl-5')+' px-2 d-flex justify-space-between align-center'"
+            @click="gotoLine(item.index)"
+          >
+            <div
+              v-if="item.section!==undefined"
+              class="py-1"
+            >
+              <span :class="$vuetify.theme.dark?'mtk1':'mtk1'">[</span> <span
+                :class="$vuetify.theme.dark?'mtk13':'mtk4'"
+              >{{ item.section }}</span> <span :class="$vuetify.theme.dark?'mtk1':'mtk1'">]</span>
```

Comment:
Class is the same no matter if dark theme is in use or not... is this what you wanted?

Reply:
It used to be different, but the latest version I found is the same, I'm not sure that, so I left this code till now.